packages/sdk/src/protocols/peer_manager.ts

```diff
@@ -0,0 +1,113 @@
+import { Peer, PeerId } from "@libp2p/interface";
+import { ConnectionManager, getHealthManager } from "@waku/core";
+import { BaseProtocol } from "@waku/core/lib/base_protocol";
+import { IHealthManager } from "@waku/interfaces";
+import { Logger } from "@waku/utils";
+import { Mutex } from "async-mutex";
+
+export class PeerManager {
+  private peers: Map<string, Peer> = new Map();
+  private healthManager: IHealthManager;
+
+  private readMutex = new Mutex();
+  private writeMutex = new Mutex();
+  private writeLockHolder: string | null = null;
+
+  public constructor(
+    private readonly connectionManager: ConnectionManager,
+    private readonly core: BaseProtocol,
+    private readonly log: Logger
+  ) {
+    this.healthManager = getHealthManager();
+    this.healthManager.updateProtocolHealth(this.core.multicodec, 0);
+  }
+
+  public getWriteLockHolder(): string | null {
+    return this.writeLockHolder;
+  }
+
+  public getPeers(): Peer[] {
+    return Array.from(this.peers.values());
+  }
+
+  public async addPeer(peer: Peer): Promise<void> {
+    return this.writeMutex.runExclusive(async () => {
+      this.writeLockHolder = `addPeer: ${peer.id.toString()}`;
+      await this.connectionManager.attemptDial(peer.id);
+      this.peers.set(peer.id.toString(), peer);
+      this.log.info(`Added and dialed peer: ${peer.id.toString()}`);
+      this.healthManager.updateProtocolHealth(
+        this.core.multicodec,
+        this.peers.size
+      );
+      this.writeLockHolder = null;
+    });
+  }
+
+  public async removePeer(peerId: PeerId): Promise<void> {
+    return this.writeMutex.runExclusive(() => {
+      this.writeLockHolder = `removePeer: ${peerId.toString()}`;
+      this.peers.delete(peerId.toString());
+      this.log.info(`Removed peer: ${peerId.toString()}`);
+      this.healthManager.updateProtocolHealth(
+        this.core.multicodec,
+        this.peers.size
+      );
+      this.writeLockHolder = null;
+    });
+  }
+
+  public async getPeerCount(): Promise<number> {
+    return this.readMutex.runExclusive(() => this.peers.size);
+  }
+
+  public async hasPeers(): Promise<boolean> {
```

Comment:
what is the diff from this `hasPeers` and `BaseProtocolSDK.hasPeers`?

Reply:
Core doesn't have any `hasPeers()`